go/tasks/v1/flytek8s/plugin_executor.go

```diff
@@ -243,6 +243,35 @@ func (e *K8sTaskExecutor) CheckTaskStatus(ctx context.Context, taskCtx types.Tas
 		PhaseVersion: taskCtx.GetPhaseVersion(),
 	}
 
+	// NOTE: To ensure objects are cleaned up, the plugins need a persistent step in addition to upstream plugin executor
+	// state machine. Once the object reaches its terminal state, we commit the completion in two steps:
+	// Round1: mark the object as deleted in state store (object's custom state)
+	// Round2: instead of regular retrieval (which may fail in this case), just delete the object
+	objStatus, terminalPhase, err := retrieveK8sObjectStatus(taskCtx.GetCustomState())
+	if err != nil {
+		logger.Warningf(ctx, "Failed to retrieve object status: %v. Error: %v",
+			taskCtx.GetTaskExecutionID().GetGeneratedName(), err)
+		return types.TaskStatusUndefined, err
+	}
+	if objStatus == k8sObjectDeleted {
+		// kill the object execution if still live
+		if e.handler.GetProperties().DeleteResourceOnAbort {
+			err = instance.kubeClient.Delete(ctx, o)
+
+			if err != nil {
+				if IsK8sObjectNotExists(err) {
+					logger.Debugf(ctx, "the k8s object %v was found to have successfully exited after completion", taskCtx.GetTaskExecutionID().GetGeneratedName())
+				} else {
+					return types.TaskStatusUndefined, err
+				}
+			} else {
+				logger.Debugf(ctx, "deleted the k8s object %v in terminal phase", taskCtx.GetTaskExecutionID().GetGeneratedName())
+			}
+		}
+		finalStatus.Phase = terminalPhase
```

Comment:
if the terminalPhase is failed
 then set error code and error message

Reply:
done